purchases/src/main/java/com/revenuecat/purchases/IdentityManager.kt

```diff
@@ -0,0 +1,85 @@
+package com.revenuecat.purchases
+
+import java.util.Locale
+import java.util.UUID
+
+internal class IdentityManager(
+    private val deviceCache: DeviceCache,
+    private val backend: Backend
+) {
+
+    val currentAppUserID: String
+        get() = deviceCache.getCachedAppUserID() ?: ""
+
+    @Synchronized
+    fun configure(appUserID: String?) {
+        val appUserIDToUse = appUserID
+            ?: deviceCache.getCachedAppUserID()
+            ?: deviceCache.getLegacyCachedAppUserID()
+            ?: generateRandomID()
+        debugLog("Identifying App User ID: $appUserIDToUse")
+        deviceCache.cacheAppUserID(appUserIDToUse)
+    }
+
+    fun identify(
+        appUserID: String,
+        onSuccess: () -> Unit,
+        onError: (PurchasesError) -> Unit
+    ) {
+        if (currentUserIsAnonymous()) {
+            debugLog("Identifying from an anonymous ID: $appUserID. An alias will be created.")
+            createAlias(appUserID, onSuccess, onError)
+        } else {
+            synchronized(this@IdentityManager) {
+                debugLog("Changing App User ID: $currentAppUserID -> $appUserID")
+                deviceCache.clearCachesForAppUserID()
+                deviceCache.cacheAppUserID(appUserID)
+            }
+            onSuccess()
+        }
+    }
+
+    fun createAlias(
+        newAppUserID: String,
+        onSuccess: () -> Unit,
+        onError: (PurchasesError) -> Unit
+    ) {
+        debugLog("Creating an alias to $currentAppUserID from $newAppUserID")
+        backend.createAlias(
+            currentAppUserID,
+            newAppUserID,
+            {
+                synchronized(this@IdentityManager) {
+                    debugLog("Alias created")
+                    deviceCache.clearCachesForAppUserID()
+                    deviceCache.cacheAppUserID(newAppUserID)
+                }
+                onSuccess()
+            },
+            onError
+        )
+    }
+
+    @Synchronized
+    fun reset() {
+        deviceCache.clearCachesForAppUserID()
+        deviceCache.cacheAppUserID(generateRandomID())
+    }
+
+    @Synchronized
+    fun currentUserIsAnonymous(): Boolean {
+        val currentAppUserIDLooksAnonymous =
```

Comment:
In the original spec, we were checking if someone was identifying with something that looked like our ID, but we dropped it because it added unnecessary complexity. If someone wants to do that, let them do that. 

I originally implemented this saving both the ID and the boolean in the SharedPreferences, but ended up being super bug prone having to update two things (and read) everytime I touched the ID